File: javaproject/app/src/main/java/com/marcosvbras/robomarket/utils/ErrorObservable.java
package com.marcosvbras.robomarket.utils;

import android.databinding.BaseObservable;

public class ErrorObservable extends BaseObservable {

    private String errorString;
    private Integer errorResource;

    public ErrorObservable() {}

    public ErrorObservable(String errorString) {
        this.errorString = errorString;
        notifyChange();
    }

    public ErrorObservable(Integer errorResource) {
        this.errorResource = errorResource;
        notifyChange();
    }

    public String get() {
        return errorString;
    }

    public Integer getIntError() { return errorResource; }

    public void set(String errorString) {
        this.errorString = errorString;
        notifyChange();
    }

    public void set(Integer errorResource) {
        this.errorResource = errorResource;
        notifyChange();
    }

    public void clear() {
        errorResource = null;
        errorString = null;
        notifyChange();
    }

    public boolean hasErrorSetted() {
        return errorString != null || errorResource != null;
    }
}

File: javaproject/app/src/main/java/com/marcosvbras/robomarket/utils/ErrorObservable.kt
package com.marcosvbras.robomarket.utils

import android.databinding.BaseObservable

class ErrorObservable() : BaseObservable() {

    constructor(errorString: String?) : this() {
        this.errorString = errorString
        notifyChange()
    }

    constructor(errorResource: Int?) : this() {
        this.errorString = errorString
        notifyChange()
    }

    var errorString: String? = null
        set(value) {
            field = value
            notifyChange()
        }

    var errorResource: Int? = null
        set(value) {
            field = value
            notifyChange()
        }

    fun clear() {
        this.errorString = null
        this.errorResource = null
        notifyChange()
    }

    fun hasError() : Boolean = this.errorResource != null || this.errorString != null

}